File: src/TaposcopeWindows/Messages/ActivityGroupMoveRequest.cs
using System;

namespace Messages
{
    public class ActivityGroupMoveRequest
    {
        // Must provide one of these, but not both.
        public Guid? Id { get; set; }
        public string MatchResponseText { get; set; }

        // These indicate where we're moving the ActivityGroup.
        public Guid? TargetParentId { get; set; }
        public string TargetParentMatchResponseText { get; set; }
        public Guid? TargetGrandparentId { get; set; }
        
        // Can provide this only if no target parent description is given.
        public bool? TargetIsUncategorized { get; set; }
    }
}

File: src/shared/api/ActivityGroupMoveRequest.ts
export interface ActivityGroupMoveRequest {
    // Should provide one of these but not both.
    id?: string
    matchResponseText?: string

    // New parent information.
    targetParentId?: string
    targetParentMatchResponseText?: string
    targetGrandparentId?: string

    // Can provide this only if new parent info is not present.
    targetIsUncategorized?: boolean
}